File: CaptureImage/app/src/main/java/com/xyz/captureimage/MainActivity.java
package com.xyz.captureimage;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Bitmap;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

public class MainActivity extends AppCompatActivity {

    private static final int CAMERA_REQUEST = 102;
    private static final int REQ_CODE = 101;
    private Button mBtnOpenCamera;
    private ImageView mIvImage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
    }

    private void initViews() {
        mBtnOpenCamera = findViewById(R.id.btnOpenCamera);
        mIvImage = findViewById(R.id.ivImage);
        mBtnOpenCamera.setOnClickListener(new View.OnClickListener() {
            @RequiresApi(api = Build.VERSION_CODES.M)
            @Override
            public void onClick(View v) {

                if (ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED
                        || ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
                    requestPermissions(new String[]{Manifest.permission.CAMERA, Manifest.permission.READ_EXTERNAL_STORAGE}, REQ_CODE);
                } else {
                    openCameraIntent();
                }
            }
        });
    }

    private void openCameraIntent() {
        Intent cameraIntent = new Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
        startActivityForResult(cameraIntent, CAMERA_REQUEST);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        if (requestCode == REQ_CODE && grantResults[0] == PackageManager.PERMISSION_GRANTED && grantResults[1] == PackageManager.PERMISSION_GRANTED) {
            openCameraIntent();
        } else {
            Toast.makeText(this, "Permission denied", Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == CAMERA_REQUEST && resultCode == Activity.RESULT_OK) {
            Bitmap photo = (Bitmap) data.getExtras().get("data");
            mIvImage.setImageBitmap(photo);
        }
    }
}

File: AndroidSQLite/app/src/main/java/com/masai/androidsqlite/MainActivity.kt
package com.masai.androidsqlite

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.masai.androidsqlite.database.AdModel
import com.masai.androidsqlite.database.DbHandler
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val dbHandler = DbHandler(this)

        val result = dbHandler.getAds()
        adsListTv.text = result


        registerBtn.setOnClickListener {
            val title = titleEt.text.toString()
            val desc = descEt.text.toString()
            val price = priceEt.text.toString()

            val adModel = AdModel()
            adModel.title = title
            adModel.desc = desc
            adModel.price = price.toInt()

            val isSuccess = dbHandler.saveAd(adModel)

            if (isSuccess) {
                Toast.makeText(this, "Ad published successfully",
                    Toast.LENGTH_SHORT).show()
                titleEt.setText("")
                descEt.setText("")
                priceEt.setText("")

            }else{
                Toast.makeText(this, "Something went wrong!!",
                    Toast.LENGTH_SHORT).show()
            }

        }

    }
}